src/emulator/extensions/postinstall.ts

```diff
@@ -0,0 +1,38 @@
+import { EmulatorRegistry } from "../registry";
+import { Emulators } from "../types";
+
+export function replaceConsoleLinks(postinstall: string): string {
+  const uiInfo = EmulatorRegistry.getInfo(Emulators.UI);
+  const uiUrl = uiInfo ? `http://${EmulatorRegistry.getInfoHostString(uiInfo)}` : "unknown";
+  let subbedPostinstall = postinstall;
+  const linkReplacements = new Map<RegExp, string>([
+    [
+      /(http[s]?:\/\/)?console\.firebase\.google\.com\/project\/[A-Za-z0-9-]+\/storage[A-Za-z0-9\/-]*(?=[\)\]\s])/,
+      `${uiUrl}/storage`,
+    ], // Storage console links
+    [
+      /(http[s]?:\/\/)?console\.firebase\.google\.com\/project\/[A-Za-z0-9-]+\/firestore[A-Za-z0-9\/-]*(?=[\)\]\s])/,
+      `${uiUrl}/firestore`,
+    ], // Firestore console links
+    [
+      /(http[s]?:\/\/)?console\.firebase\.google\.com\/project\/[A-Za-z0-9-]+\/database[A-Za-z0-9\/-]*(?=[\)\]\s])/,
+      `${uiUrl}/database`,
+    ], // RTDB console links
+    [
+      /(http[s]?:\/\/)?console\.firebase\.google\.com\/project\/[A-Za-z0-9-]+\/authentication[A-Za-z0-9\/-]*(?=[\)\]\s])/,
+      `${uiUrl}/auth`,
+    ], // Auth console links
+    [
+      /(http[s]?:\/\/)?console\.firebase\.google\.com\/project\/[A-Za-z0-9-]+\/functions[A-Za-z0-9\/-]*(?=[\)\]\s])/,
+      `${uiUrl}/logs`,
```

Comment:
is there a way we can keep the ui repo and this in sync? so a route change doesn't break us.

an IfThisThenThat lint could work except they aren't in the same repo... 

maybe throw a comment in the emulatorUI repo?

Reply:
I ended up refactoring this to be more similar to the way we generate links for the table that is printed out at the end of emulators:start. Hopefully that makes them a little less likely to break, since those links are very obvious + well established.